### PR TITLE
feat: Add geography support

### LINKED ISSUE #136: Add support for the BigQuery geography type.
The BQ geography type isn't supported.

It should be.  It would be great to get suggestions on how it should behave. :)

### PR DESCRIPTION
**Pull request by jimfulton**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-sqlalchemy/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #136 🦕

### CODE REVIEW

**Review by tswast — CHANGES_REQUESTED** [review]

File: docs/geography.rst
Comment (jimfulton) [reply docs/geography.rst:44]: It's true because I made it so. :)  I'm arranging for the coercion to happen at the SQLAlchemy level. See https://github.com/jimfulton/python-bigquery-sqlalchemy/blob/geography2/pybigquery/geography.py#L98-L108

File: noxfile.py
Comment (jimfulton) [reply noxfile.py:101]: OK, I'll try that. I'm not sure what it's going to do with compliance tests.

File: pybigquery/geography.py
Comment (jimfulton) [reply pybigquery/geography.py:27]: Sure

File: pybigquery/geography.py
Comment (jimfulton) [reply pybigquery/geography.py:30]: OK.

File: pybigquery/geography.py
Comment (jimfulton) [reply pybigquery/geography.py:42]: OK

File: pybigquery/geography.py
Comment (jimfulton) [reply pybigquery/geography.py:58]: OK. Do you want to make it "Geography"?

File: setup.py
Comment (jimfulton) [reply setup.py:78]: I don't think so.  If they're using alembic, they're install it themselves. pybigquery of more of a plugin for alembic than the other way around.

File: tests/system/test_geography.py
Comment (jimfulton) [reply tests/system/test_geography.py:24]: No, IMO.  I don't want to hide an import problem.  I know that we install geoalchemy2 for Python 3.9 and I want to fail if the import fails for Python 3.9.

File: pybigquery/geography.py
Comment (tswast) [reply pybigquery/geography.py:58]: I suppose most / all of our other type classes are all caps too. Maybe we document it in some contributing docs?

File: pybigquery/geography.py
Comment (tswast) [inline pybigquery/geography.py:98]: Could we get a docstring here? I'd like some more context about how this function is expected to be used.

File: tests/system/test_geography.py
Comment (tswast) [reply tests/system/test_geography.py:24]: OK

File: tests/unit/test_geography.py
Comment (tswast) [inline tests/unit/test_geography.py:1]: Needs license header.

File: pybigquery/geography.py
Comment (jimfulton) [reply pybigquery/geography.py]: done

File: pybigquery/geography.py
Comment (jimfulton) [reply pybigquery/geography.py:30]: done

File: pybigquery/geography.py
Comment (jimfulton) [reply pybigquery/geography.py:50]: done

File: pybigquery/geography.py
Comment (jimfulton) [reply pybigquery/geography.py:86]: I suggest adding a comment here: 
https://github.com/googleapis/python-bigquery-sqlalchemy/blob/080eda16933be11b156e85d5a83225f5bb8377dc/pybigquery/sqlalchemy_bigquery.py#L128

Although, I'm not sure what to say. :)

Maybe:

`# By convention, dialect-provided types are spelled with all upper case.`

?

File: tests/unit/test_geography.py
Comment (jimfulton) [reply tests/unit/test_geography.py:20]: Oops, fixed

**Comment by jimfulton** [comment]
Note that I refactored the docs a bit to make navigation easier and to add api documentation for GEOGRAPHY, WKT and WKB.

I also dropped documentation for the old api for doing fry-run queries, since we decided we were going to drop that api.

File: noxfile.py
Comment (jimfulton) [reply noxfile.py]: Done.  Unfortunately, as expected, this didn't work for compliance tests.

File: noxfile.py
Comment (jimfulton) [reply noxfile.py]: Done

File: setup.py
Comment (jimfulton) [reply setup.py]: Well, we did to leverage the automatic extras machinery.

File: pybigquery/geography.py
Comment (jimfulton) [reply pybigquery/geography.py:86]: Done

File: tests/system/test_geography.py
Comment (jimfulton) [reply tests/system/test_geography.py]: I changed my mind.   Test code should be independent of nox.

If we fail to run some tests due to nox misconfiguration, we'll catch it with coverage.

File: docs/geography.rst
Comment (tswast) [inline docs/geography.rst:30]:
```suggestion
      geog = Column(GEOGRAPHY)
```

File: docs/geography.rst
Comment (tswast) [inline docs/geography.rst:85]: Let's include an example of the `pip` command in case folks aren't familiar with the "extra" term.

```suggestion
`shapely`.

.. code-block:: console

    pip install 'pybigquery[geography]'
```

File: owlbot.py
Comment (tswast) [inline owlbot.py:31]: Awesome!

File: owlbot.py
Comment (tswast) [inline owlbot.py:106]: Maybe add a link to query retry issue/pr?

File: pybigquery/geography.py
Comment (tswast) [inline pybigquery/geography.py:132]:
```suggestion
        This is not a user-facing method.
```

**Review by tswast — COMMENTED** [review]
Looks good! A few minor comments.

File: docs/geography.rst
Comment (jimfulton) [reply docs/geography.rst:30]: That's what I get for having docs that aren't automatically tested.  I'll convert this to use a `snippets.py`, following the `python-bigquery` example.

File: docs/geography.rst
Comment (jimfulton) [reply docs/geography.rst:85]: OK, will do.

File: owlbot.py
Comment (jimfulton) [reply owlbot.py:106]: OK

**Comment by jimfulton** [comment]
I added the "do not merge" label.  We won't merge this until the after the rename branch lands.

File: owlbot.py
Comment (jimfulton) [reply owlbot.py]: Done

**Review by tswast — APPROVED** [review]
I like it!

I wonder if we want to merge this before the rename just to make the docfx session happy?

**Comment by jimfulton** [comment]
> I wonder if we want to merge this before the rename just to make the docfx session happy?

I'm working on a branch that combines the rename and geography (and snippets to be used in the geography doc).

I suggest we decide next week whether to wait for Dan to fix docfx or to include geography in alpha 1.

**Comment by jimfulton** [comment]
I merged the rename branch into this. Maybe that was a mistake.

We'll either want to merge the rename branch first, at which point this will be smaller, or we'll merge this.

**Comment by jimfulton** [comment]
Everything has been reviewed here except: https://github.com/googleapis/python-bigquery-sqlalchemy/pull/228/commits/6343bee73490b68b513698d0c9a79a99ceaa8438, which changes the geography documentation to use snippets, which found a number of bugs. :)  I also improved one of the query examples.

I tried to format the query examples to be more readable in the documentation, but black wouldn't let me. :(

**Comment by jimfulton** [comment]
@tswast I'm going to assume you still approve unless I hear otherwise. :smiley: 

I'm thinking of waiting to merge this until after 1.0.0-a1 is released, unless you want me to include these changes in a1.

**Review by tswast — APPROVED** [review]
Looks good!

I'm okay with waiting to merge this until after the a1 release.

**Pull request merged**
